lib/app.js

```diff
@@ -24,21 +24,89 @@ app.use('/api', ensureAuth);
 // and now every request that has a token in the Authorization header will have a `req.userId` property for us to see who's talking
 app.get('/api/test', (req, res) => {
   res.json({
-    message: `in this proctected route, we get the user's id like so: ${req.userId}`
+    message: `in this protected route, we get the user's id like so: ${req.dessertsId}`
   });
 });
 
-app.get('/animals', async(req, res) => {
+app.post('/desserts', async (req, res) => {
   try {
-    const data = await client.query('SELECT * from animals');
-    
-    res.json(data.rows);
-  } catch(e) {
-    
+    const data = await client.query(`
+    INSERT INTO desserts 
+    (name, icing, type) 
+    VALUES ($1, $2, $3)
+    RETURNING * `, [
+      req.body.name,
+      req.body.icing,
+      req.body.type
+    ]);
+    res.json(data.rows[0]);
+  } catch (e) {
+
     res.status(500).json({ error: e.message });
   }
 });
 
-app.use(require('./middleware/error'));
+app.get('/desserts/:id', async (req, res) => {
+  const id = req.params.id;
+  try {
+    const data = await client.query('SELECT * from desserts/:id', [id]);
```

Comment:
You were SUPER close here too! The query for get by id just needs a `WHERE` clause to tell SQL which ID to look for. Also, while your route endpoint (the part in quotes before `async` has /:id in it, your SQL query will not. Remember, SQL is different than your API. To getById, you'll want to search your DESSERTS table for the item WHERE id is equal to the ID supplied in the URL params.